avaje-jex/src/main/java/io/avaje/jex/core/RoutingHandler.java

```diff
@@ -36,6 +36,7 @@ public void handle(HttpExchange exchange) {
 
     if (route == null) {
       var ctx = new JdkContext(mgr, exchange, uri, Set.of());
+      ctx.setMode(Mode.EXCHANGE);
```

Comment:
I think we can move this into the constructor. Will merge this in but look to adjust later.